### PR TITLE
[DEV-1948] api/target: add string and datetime accessors

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Add the string and date time transformation accessors into the `Target` API object.

## Related Issue
https://featurebyte.atlassian.net/browse/DEV-1948

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/core/accessor/target_string.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        Returns
+        -------
+        Target
+            A new Target object.
+
+        Examples
+        --------
+        Compute the length of each string element in the ProductGroupLookup target:
+
+        >>> target = catalog.get_target("ProductGroupLookup")  # doctest: +SKIP
```
Comment (jevonyeoh) [inline featurebyte/core/accessor/target_string.py:56]: i've skipped these for now. will unskip when i add the lookup target functionality.

File: featurebyte/core/accessor/target_string.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+            Characters to remove, whitespace is used by default.
+
+        Returns
+        -------
+        Target
+            A new Target object.
+
+        Examples
+        --------
+        Remove leading and trailing "M" characters from the Title column:
```
Comment (YS-L) [inline featurebyte/core/accessor/target_string.py:113]: The term `column` or `Series` are sometimes still referenced in these docstrings. Is that common in the existing docstrings?

File: featurebyte/core/accessor/target_datetime.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        Returns
+        -------
+        Target
+            Target containing the year component values
+
+        Examples
+        --------
+        Compute the year component of a timestamp column:
+
+        >>> target = catalog.get_target("target_customer_amount_7d")  # doctest: +SKIP
```
Comment (YS-L) [inline featurebyte/core/accessor/target_datetime.py:80]: Would be a bit better if the target referenced here is of a timestamp type.

**Review by YS-L — APPROVED** [review]

File: featurebyte/core/accessor/target_string.py (hunk 2 above)
Comment (jevonyeoh) [reply featurebyte/core/accessor/target_string.py:113]: yea, some of the feature docstrings reference that right now. let me clean it up.

**Pull request merged**